dolphin/api/v1/storages.py

```diff
@@ -109,16 +109,18 @@ def create(self, req, body):
             raise exc.HTTPBadRequest(explanation=msg)
 
         try:
-            storage = self.driver_manager.register_storage(ctxt, access_info_dict)
-            storage = db.storage_create(context, storage)
+            storage = self.driver_manager.register_storage(ctxt,
+                                                           access_info_dict)
 
             # Need to encode the password before saving.
             access_info_dict['storage_id'] = storage['id']
```

Comment:
should we make sure storage['id'] existes before updating DB, beacuse this field is used in is_registered to make sure access_info and storage are in sysnc.

Reply:
no need here because it's generated in driver manager.  and it's under try catch, will be catched if 'id' is not in storage.

Reply:
What I mean is  storage['id'] may be initialized by driver but not filled value, Is'nt it possible that you get a None or '' in storage['id']?